dace/transformation/passes/constant_propagation.py

```diff
@@ -69,40 +74,85 @@ def apply_pass(self, sdfg: SDFG, _, initial_symbols: Optional[Dict[str, Any]] =
         if not initial_symbols and not self.should_apply(sdfg):
             result = {}
         else:
-            # Trace all constants and symbols through states
-            per_state_constants: Dict[SDFGState, Dict[str, Any]] = self.collect_constants(sdfg, initial_symbols)
```

Comment:
I find it weird that this method has been somewhat inlined. What is the rationale behind this?

Reply:
It wasn't really inlined, the functionality of collecting constants is still contained inside the (now renamed) method `_collect_constants_for_region`, which operates recursively. The only thing that was inlined is the generation of the `arrays` set, which collects all nested data names. This was only inlined because it needs to be done only once per SDFG (and thus per pass application), and not in every recursive call to `_collect_constants_for_region`.